### ISSUE TITLE
New architecture support

### ISSUE THREAD

**Issue by fobos531** [stripe/stripe-terminal-react-native#848]
**Is your feature request related to a problem? Please describe.**

RN 0.76 is officially out and it enables new architecture by default.

**Describe the solution you'd like**

This library should properly support new architecture, otherwise users are forced to disable new architecture.

**Comment by nazli-stripe** [comment]
We are aiming to add new architecture support in early 2025.

**Comment by Brunowilliang** [comment]
@nazli-stripe, I've tested it and the terminal is working with the new architecture. Is there a problem that justifies opening this issue?

**Comment by nazli-stripe** [comment]
@Brunowilliang we haven't tested it so don't want to officially claim support for it, and it is in our plans to upgrade our underlying RN version to 0.76. Everything was working fine in your testing?

@fobos531 did you run into issues?

**Comment by Brunowilliang** [comment]
yes, it worked for me @nazli-stripe, in order for it to work I had to adjust a few things in ‘expo-build-properties’, but it worked...

```json:
	[
		'expo-build-properties',
		{
			android: {
				minSdkVersion: 26,
				compileSdkVersion: 35,
				targetSdkVersion: 35,
				packagingOptions: {
					pickFirst: ['org/bouncycastle/x509/CertPathReviewerMessages*'],
					exclude: [
						'org/bouncycastle/pqc/crypto/picnic/lowmcL5.bin.properties',
						'org/bouncycastle/pqc/crypto/picnic/lowmcL3.bin.properties',
						'org/bouncycastle/pqc/crypto/picnic/lowmcL1.bin.properties',
					],
				},
			},
			ios: {
				deploymentTarget: '15.1',
			},
		},
	],
```

I don't know if I can do that, but I managed to make it work 🤣

**Comment by Salman-Usmani** [comment]
I faced an issue with CardForm (which I must use to get a token), when newArch is enabled the onFormComplete callback doesn’t trigger, and the UI behaves inconsistently—sometimes asking for a ZIP code and sometimes not.
hope it helps and we get to see newArch soon...

**Comment by Salman-Usmani** [comment]
@nazli-stripe you are right, i am sorry.

**Comment by Noitidart** [comment]
> [
> 		'expo-build-properties',
> 		{
> 			android: {
> 				minSdkVersion: 26,
> 				compileSdkVersion: 35,
> 				targetSdkVersion: 35,
> 				packagingOptions: {
> 					pickFirst: ['org/bouncycastle/x509/CertPathReviewerMessages*'],
> 					exclude: [
> 						'org/bouncycastle/pqc/crypto/picnic/lowmcL5.bin.properties',
> 						'org/bouncycastle/pqc/crypto/picnic/lowmcL3.bin.properties',
> 						'org/bouncycastle/pqc/crypto/picnic/lowmcL1.bin.properties',
> 					],
> 				},
> 			},
> 			ios: {
> 				deploymentTarget: '15.1',
> 			},
> 		},
> 	],

Thank you! This worked for me! Did you need the `ios.deploymentTarget` increase? I didn't need that part.
